src/lib/query/sriov.rs

```diff
@@ -97,6 +99,12 @@ pub(crate) fn get_sriov_info(
         IfaceType::Infiniband => INFINIBAND_ALEN,
         _ => MAX_ADDR_LEN,
     };
+    if let Some(num) = get_num_vfs(pf_iface_name) {
+        sriov_info.num_vfs = num;
+    }
+    if let Some(autoprobe) = is_drivers_autoprobe(pf_iface_name) {
+        sriov_info.drivers_autoprobe = autoprobe;
+    }
```

Comment:
As I said, the values must always be set to the proper value, either `None` or `Some(value)`.